src/app/Controllers/ActorsController.cs

```diff
@@ -28,49 +31,41 @@ public ActorsController(ILogger<ActorsController> logger, IDAL dal)
         }
 
         /// <summary>
-        /// Returns a JSON array of Actor objects
+        /// Returns a JSON array of Actor objects based on query parameters
         /// </summary>
-        /// <param name="q">(optional) The term used to search Actor name</param>
-        /// <param name="pageNumber">1 based page index</param>
-        /// <param name="pageSize">page size (1000 max)</param>
-        /// <response code="200">JSON array of Actor objects or empty array if not found</response>
+        /// <param name="actorQueryParameters"></param>
         [HttpGet]
-        public async Task<IActionResult> GetActorsAsync([FromQuery] string q, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
+        public async Task<IActionResult> GetActorsAsync([FromQuery] ActorQueryParameters actorQueryParameters)
         {
-            string method = GetMethodText(q, pageNumber, pageSize);
-
-            // validate query string parameters
-            var result = ParameterValidator.Common(HttpContext?.Request?.Query, q, pageNumber, pageSize, method, logger);
-            if (result != null)
-            {
-                return result;
-            }
+            _ = actorQueryParameters ?? throw new ArgumentNullException(nameof(actorQueryParameters));
 
             // convert to zero based index
-            pageNumber = pageNumber > 1 ? pageNumber - 1 : 0;
+            actorQueryParameters.PageNumber = actorQueryParameters.PageNumber > 1 ? actorQueryParameters.PageNumber - 1 : 0;
 
-            return await ResultHandler.Handle(dal.GetActorsAsync(q, pageNumber * pageSize, pageSize), method, Constants.ActorsControllerException, logger).ConfigureAwait(false);
+            return await ResultHandler.Handle(dal.GetActorsAsync(
+                    actorQueryParameters.Q,
+                    actorQueryParameters.PageNumber * actorQueryParameters.PageSize,
+                    actorQueryParameters.PageSize),
+                    GetMethodText(actorQueryParameters),
+                    Constants.ActorsControllerException,
+                    logger)
+                .ConfigureAwait(false);
         }
 
         /// <summary>
         /// Returns a single JSON Actor by actorId
         /// </summary>
-        /// <param name="actorId">The actorId</param>
+        /// <param name="actorIdParameter">The actorId</param>
         /// <response code="404">actorId not found</response>
         [HttpGet("{actorId}")]
-        public async Task<IActionResult> GetActorByIdAsync(string actorId)
+        public async Task<IActionResult> GetActorByIdAsync([FromRoute] ActorIdParameter actorIdParameter)
```

Comment:
since this is part of the URL, it doesn't really feel like a "parameter"

Reply:
If I changed this to `actorId` it would conflict with the route name. When you use model binding you can't have a parameter name the same as the route template. If you have a suggested name I can incorporate it but I feel this is fine.